src/display/ui_state.rs

```diff
@@ -110,10 +110,13 @@ impl UIState {
         let mut connections: BTreeMap<Connection, ConnectionData> = BTreeMap::new();
         let mut total_bytes_downloaded: u128 = 0;
         let mut total_bytes_uploaded: u128 = 0;
+
+        let mut seen_connections = HashSet::new();
         for state in self.utilization_data.iter().rev() {
             let connections_to_procs = &state.connections_to_procs;
             let network_utilization = &state.network_utilization;
             for (connection, connection_info) in &network_utilization.connections {
+                let connection_previously_seen = seen_connections.contains(connection);
```

Comment:
This can be replaced with 
```
let connection_previously_seen = !seen_connections.insert(connection);
```

And the insert statement at the bottom can be removed.

Reply:
Ah, right! Thanks for this :)